### PR TITLE
Initial roadmap submission

### PR DESCRIPTION
**Pull request by jeremyrickard**
Initial roadmap and update to README to link to it.

### CODE REVIEW

**Comment by krancour** [comment]
No CI needed on this PR.

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 [Open Service Broker](https://www.openservicebrokerapi.org/)-compatible API
 server that provisions managed services in the Microsoft Azure public cloud.
 
-|![](https://upload.wikimedia.org/wikipedia/commons/thumb/1/17/Warning.svg/150px-Warning.svg.png) | This software is under heavy development. Releases observe [semantic versioning](https://semver.org), but the project is in an ALPHA status and no assurances are made regarding backwards compatibility or stability. All releases prior to v1.0.0 will remain subject to the possibility of breaking changes when the MINOR version number has been incremented. |
+|![](https://upload.wikimedia.org/wikipedia/commons/thumb/1/17/Warning.svg/150px-Warning.svg.png) | This software is under heavy development. Releases observe [semantic versioning](https://semver.org), but the project is in an ALPHA status and no assurances are made regarding backwards compatibility or stability. All releases prior to v1.0.0 will remain subject to the possibility of breaking changes when the MINOR version number has been incremented. Please see the offical [roadmap](docs/roadmap.md) for more information. |
```
Comment (krancour) [inline README.md]: Great spot to reference the roadmap. 👍

**Comment by krancour** [comment]
Unless @arschles wants to tweak wording at all (I think these are mostly his own words anyway), I am LGTM on this.

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+* Database Schemas
+
+See below for more information concerning stability plans for each.
+
+# Services and Plans
+
+Services/Plans have three stability tiers:
+
+- `experimental` - We have an idea for a new service that we want to support, but don't have a good idea how it should look. We are essentially “throwing something against the wall” to see if it'll stick. Experimental services may be radically changed or removed at any time
+- `preview` - We have a better understanding how a service should look, but we don't yet guarantee backward compatibility. We do guarantee that this service won't go back to `experimental`, so we are committing to making it exist in some shape or form
```
Comment (arschles) [inline docs/roadmap.md]: can we make a more specific guarantee here? for example, are we committing to working on it until it becomes stable in some shape or form?

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+- SQL Server: `preview` by the end of February 2018
+- MySQL: `preview` by the end of February 2018
+- PostgreSQL: `preview` by the end of February 2018
+
+We aren't yet setting timelines on other service stabilities.
+
+# Database Schemas
+
+There are two tiers of database schemas to consider:
```
Comment (arschles) [inline docs/roadmap.md]: Can you explain here why database schemas are important to consider for stability?

File: docs/roadmap.md (hunk 2 above)
Comment (krancour) [reply docs/roadmap.md]: Not sure if we can commit to that in all cases. It's just occurred to me that some of the Azure-managed services that our modules can provision are _themselves_ in preview. For instance, Azure Database for MySQL || PostgreSQL are themselves both in preview. In such cases, we probably cannot make any guarantees.

Comment (jeremyrickard) [reply docs/roadmap.md]: Can we commit to maturing the modules to match the maturity level of the service itself? I.e. we won't make Azure DB for MySQL or PostgreSQL stable until the underlying service is generally available and out of preview?

**Comment by seanmck** [comment]
Same - LGTM.

**Comment by jeremyrickard** [comment]
Skipping CI and merging with LGTM from @seanmck @arschles and @krancour

**Pull request merged**